src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -88,7 +88,7 @@ protected void onMessage(Context context, Intent intent) {
         Bundle extras = intent.getExtras();
         if (extras != null) {
             // if we are in the foreground, just surface the payload, else post it to the statusbar
-            if (PushPlugin.isInForeground()) {
+            if (!FORCE_SHOW && PushPlugin.isInForeground()) {
```

Comment:
Shouldn't you be getting this option in a way similar to this?
```java
SharedPreferences prefs = context.getSharedPreferences(PushPlugin.COM_ADOBE_PHONEGAP_PUSH, Context.MODE_PRIVATE);
boolean force_show = prefs.getBoolean(FORCE_SHOW, false);
```

Accessing FORCE_SHOW directly is only targeting the `String` defined in PushConstants, and a `String` is always true'ish, thus making it not an option to decide on.